pkg/tsdb/azuremonitor/types.go

```diff
@@ -127,3 +130,41 @@ type logJSONQuery struct {
 		Workspace    string `json:"workspace"`
 	} `json:"azureLogAnalytics"`
 }
+
+// InsightsDimensions will unmarshal from a JSON string, or an array of strings,
+// into a string array. This exists to support an older query format which is updated
+// when a user saves the query or it is sent from the front end, but may not be when
+// alerting fetches the model.
+type InsightsDimensions []string
+
+// UnmarshalJSON fulfills the json.Unmarshaler interface type.
+func (s *InsightsDimensions) UnmarshalJSON(data []byte) error {
+	*s = InsightsDimensions{}
+	if string(data) == "null" || string(data) == "" {
+		return nil
+	}
+	if strings.ToLower(string(data)) == "none" {
+		return nil
+	}
+	if data[0] == '[' {
+		var sa []string
+		err := json.Unmarshal(data, &sa)
+		if err != nil {
+			return err
+		}
+		*s = InsightsDimensions(sa)
+		return nil
+	}
+
+	var str string
+	err := json.Unmarshal(data, &str)
+	if err != nil {
+		return fmt.Errorf("could not parse as string or array: %v", s)
```

Comment:
This makes more sense to me (doesn't make sense to output `s` in the error?):
```suggestion
		return fmt.Errorf("could not parse %q as string or array: %w", string(data), err)
```